src/fsrs/default.ts

```diff
@@ -5,44 +5,44 @@ import { clamp } from './help'
 
 export const default_request_retention = 0.9
 export const default_maximum_interval = 36500
-export const default_w = [
+export const default_w = Object.freeze([
   0.40255, 1.18385, 3.173, 15.69105, 7.1949, 0.5345, 1.4604, 0.0046, 1.54575,
   0.1192, 1.01925, 1.9395, 0.11, 0.29605, 2.2698, 0.2315, 2.9898, 0.51655,
   0.6621,
-]
+]) as number[]
```

Comment:
```suggestion
]) as readonly number[]
```
Is this needed? Fails some tests.

Reply:
I don’t want to upgrade the version to 5.0, writing readonly number[] would be a feature

Reply:
The original design does not allow modification of default_xxx, but it should allow normal data reading. Alternatively, can we modify the FSRS input parameters to allow `readonly number[]`?

Reply:
> Would this cause problems after this update?

It will throw an error. Ok,I will allow passing `readonly number[]` for w.

Reply:
> would there not be an error with `weights[3] = 100` also?

Yes… but the IDE will prompt that modification is not allowed

Reply:
![image](https://github.com/user-attachments/assets/5051ce9e-e47f-40ac-b9cb-f0b92db11a0c)
but that error isnt there before the update?

Reply:
> but that error isnt there before the update right?

The `const weights = default_w` still references default_w. Essentially, it modifies default_w, which is not the expected behavior.

```ts
const weights = default_w
weights[3] = 100
console.log(weights[3])
console.log(default_w[3])
```
![image](https://github.com/user-attachments/assets/f283d159-72f4-416c-ac89-c331941aadb0)

Reply:
> The const weights = default_w still references default_w. Essentially, it modifies default_w, which is not the expected behavior.

Yes, what I'm saying is that before the update you *could* do that, so if someone did do that their project will no longer function if they update this package?

Reply:
I know, I hope they can realize the issue through IDE error reporting.
@L-M-Sherlock @asukaminato0721 What do you think about it?

Reply:
I think IDE error reporting is sufficient. Since they even use ts, so they are likely to have an IDE to do suggestions.

Reply:
Oh wait, assinging to object.freeze doesnt cause an error? 😅😅😅